ansible_base/authenticator_plugins/saml.py

```diff
@@ -0,0 +1,191 @@
+        ),
+    )
+    SP_EXTRA = serializers.JSONField(
+        default={"requestedAuthnContext": False},
+        help_text=_("A dict of key value pairs to be passed to the underlying python-saml Service Provider configuration setting."),
+    )
+    SECURITY_CONFIG = serializers.JSONField(
+        default={},
+        help_text=_(
+            "A dict of key value pairs that are passed to the underlying python-saml security setting https://github.com/onelogin/python-saml#settings"
+        ),
+    )
+    EXTRA_DATA = serializers.ListField(
+        default=[],
+        help_text=_("A list of tuples that maps IDP attributes to extra_attributes. Each attribute will be a list of values, even if only 1 value."),
+    )
+    IDP_URL = URLField(
+        allow_null=False,
+        help_text=_("The URL to redirect the user to for login initiation."),
+    )
+    IDP_X509_CERT = PublicCert(
+        allow_null=False,
+        help_text=_("The public cert used for secrets coming from the IdP."),
+    )
+    IDP_ENTITY_ID = serializers.CharField(
+        allow_null=False,
+        help_text=_("The entity ID returned in the assertion."),
+    )
+    IDP_GROUPS = serializers.CharField(
+        help_text=_("The field in the assertion which represents the users groups."),
+    )
+    IDP_ATTR_EMAIL = serializers.CharField(
+        allow_null=False,
+        help_text=_("The field in the assertion which represents the users email."),
+    )
+    IDP_ATTR_USERNAME = serializers.CharField(
+        help_text=_("The field in the assertion which represents the users username."),
+    )
+    IDP_ATTR_LAST_NAME = serializers.CharField(
+        allow_null=False,
+        help_text=_("The field in the assertion which represents the users last name."),
+    )
+    IDP_ATTR_FIRST_NAME = serializers.CharField(
+        allow_null=False,
+        help_text=_("The field in the assertion which represents the users first name."),
+    )
+    IDP_ATTR_USER_PERMANENT_ID = serializers.CharField(
+        help_text=_("The field in the assertion which represents the users permanent id (overrides IDP_ATTR_USERNAME)"),
+    )
+
+    def validate(self, attrs):
+        # attrs is only the data in the configuration field
+        errors = {}
+        # pull the cert_info out of the existing object (if we have one)
+        cert_info = {
+            "SP_PRIVATE_KEY": getattr(self.instance, 'configuration', {}).get('SP_PRIVATE_KEY', None),
+            "SP_PUBLIC_CERT": getattr(self.instance, 'configuration', {}).get('SP_PUBLIC_CERT', None),
+        }
+
+        # Now get the cert_info out of the passed in attrs (if there is any)
+        for cert_type in cert_info.keys():
+            item = attrs.get(cert_type, None)
+            if item and item == ENCRYPTED_STRING and not self.instance:
+                # Catch a case where we got input but it was ENCRYPTED and we don't have an object yet
+                errors[cert_type] = f"Can not be {ENCRYPTED_STRING} on creation"
```

Comment:
Does the authenticator serializer already handle this case? It seems like maybe we can drop this check entirely?

https://github.com/ansible/django-ansible-base/blob/b2509574cd6b17a1e9c095c412a248278877129a/ansible_base/serializers/authenticator.py#L89C13-L89C13

Reply:
In fact this happens before the authenticator config's validate is called, so I think this check in saml.py is dead code currently.

Reply:
Agreed, removed. I wonder if I started my local testing before that change merged? 🤷